File: P929 Unique Email Addresses/Solution.java
class Solution {
    // Time Complexity: O(C), where C is the total content of emails.
    // Space Complexity: O(C).
    public int numUniqueEmails(String[] emails) {
        Set<String> emailSet = new HashSet();
        for(int i = 0;i < emails.length; i++) {
            String[] emailParts = emails[i].split("@");
            String local = emailParts[0].replace(".", "");
            if(local.contains("+")) {
                local = local.substring(0, local.indexOf("+"));
            }
            emailSet.add(local + "@" + emailParts[1]);
        }
        return emailSet.size();
    }
}

File: P929 Unique Email Addresses/solution.py
class Solution:
    # Time Complexity: O(C), where C is the total content of emails.
    # Space Complexity: O(C).
    def numUniqueEmails(self, emails: List[str]) -> int:
        email_set = set()
        for email in emails:
            local, domain = email.split("@")
            local = local.replace(".", "")
            if "+" in local:
                local = local.split("+")[0]
            email_set.add(local + "@" + domain)
        return len(email_set)